### PR TITLE
planner: update binding's original_sql when upgrading

### LINKED ISSUE #44298: planner: bindings for `in (?)` can not match `in (?, ... ?)`
## Enhancement
See the case below, the binding for `... a in (1, 2, 3)` can not match queries with only 1 element in the in-list `... a in (1)` :
```
mysql> create table t (a int);
Query OK, 0 rows affected (0.09 sec)

mysql> create binding for select * from t where a in(1, 2, 3) using select * from t where a in (1, 2, 3);
Query OK, 0 rows affected (0.01 sec)

mysql> select * from t where a in (1, 2,3);
Empty set (0.01 sec)

mysql> select @@last_plan_from_binding;
+--------------------------+
| @@last_plan_from_binding |
+--------------------------+
|                        1 |
+--------------------------+
1 row in set (0.00 sec)

mysql> select * from t where a in (1, 2);
Empty set (0.01 sec)

mysql> select @@last_plan_from_binding;
+--------------------------+
| @@last_plan_from_binding |
+--------------------------+
|                        1 |
+--------------------------+
1 row in set (0.00 sec)

mysql> select * from t where a in (1);
Empty set (0.00 sec)

mysql> select @@last_plan_from_binding;
+--------------------------+
| @@last_plan_from_binding |
+--------------------------+
|                        0 |
+--------------------------+
1 row in set (0.00 sec)
```

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: close #44298

Problem Summary: planner: update binding's original_sql when upgrading

### What is changed and how it works?

Following https://github.com/pingcap/tidb/pull/46896, update binding's original_sql when upgrading to make it compatible.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: bindinfo/handle.go [hunk 1]
```diff
@@ -136,18 +136,20 @@ func (h *BindHandle) Reset(ctx sessionctx.Context) {
 func (h *BindHandle) Update(fullLoad bool) (err error) {
 	h.bindInfo.Lock()
 	lastUpdateTime := h.bindInfo.lastUpdateTime
-	updateTime := lastUpdateTime.String()
+	timeCondition := fmt.Sprintf("WHERE update_time>'%s'", lastUpdateTime.String())
 	if fullLoad {
-		updateTime = "0000-00-00 00:00:00"
```
Comment (qw4990) [inline bindinfo/handle.go:141]: `"0000-00-00 00:00:00"` is invalid.

File: bindinfo/handle.go [hunk 2]
```diff
@@ -136,18 +136,20 @@ func (h *BindHandle) Reset(ctx sessionctx.Context) {
...
+		timeCondition = ""
```
Comment (hawkingrei) [inline bindinfo/handle.go:141]:
```
var timeCondition string
if !fullLoad {
    timeCondition = fmt.Sprintf("WHERE update_time>'%s'", lastUpdateTime.String())
}
```

File: bindinfo/handle.go [hunk 3]
```diff
@@ -136,18 +136,20 @@ func (h *BindHandle) Reset(ctx sessionctx.Context) {
...
 	exec := h.sctx.Context.(sqlexec.RestrictedSQLExecutor)
 
 	ctx := kv.WithInternalSourceType(context.Background(), kv.InternalTxnBindInfo)
 	// No need to acquire the session context lock for ExecRestrictedSQL, it
 	// uses another background session.
-	rows, _, err := exec.ExecRestrictedSQL(ctx, nil, `SELECT original_sql, bind_sql, default_db, status, create_time, update_time, charset, collation, source, sql_digest, plan_digest
-	FROM mysql.bind_info WHERE update_time > %? ORDER BY update_time, create_time`, updateTime)
+	selectStmt := fmt.Sprintf(`SELECT original_sql, bind_sql, default_db, status, create_time,
+       update_time, charset, collation, source, sql_digest, plan_digest FROM mysql.bind_info
+       %s ORDER BY update_time, create_time`, timeCondition)
```
Comment (AilinKid) [inline bindinfo/handle.go:151]: miss where ?

File: bindinfo/handle.go (hunk 2 above)
Comment (qw4990) [reply bindinfo/handle.go]: updated

**Review by hawkingrei — APPROVED** [review]

**Review by AilinKid — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Comment by easonn7** [comment]
/approve

there is no sysvar change.

**Pull request merged**